src/fallbackServiceStub.ts

```diff
@@ -91,14 +92,18 @@ export function generateServiceStub(
         : new NodeAbortController();
       const cancelSignal = cancelController.signal as AbortSignal;
       let cancelRequested = false;
-
-      const fetchParameters = requestEncoder(
-        rpc,
-        protocol,
-        servicePath,
-        servicePort,
-        request
-      );
+      let fetchParameters: FetchParameters;
+      try {
+        fetchParameters = requestEncoder(
+          rpc,
+          protocol,
+          servicePath,
+          servicePort,
+          request
+        );
+      } catch (err) {
+        return Promise.resolve(callback(err));
```

Comment:
This seems to be the right place to catch it.

I wonder if just doing `if (callback) { callback(err); } return;` will do the trick without explicitly creating a promise here. Can you try that?